packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -1478,4 +1389,67 @@ export class ZipFS extends BasePortableFakeFS {
     const interval = setInterval(() => {}, 24 * 60 * 60 * 1000);
     return {on: () => {}, close: () => {clearInterval(interval);}};
   }
+
+  watchFile(p: PortablePath, cb: WatchFileCallback): StatWatcher;
+  watchFile(p: PortablePath, opts: WatchFileOptions, cb: WatchFileCallback): StatWatcher;
+  watchFile(p: PortablePath, a: WatchFileOptions | WatchFileCallback, b?: WatchFileCallback) {
+    const resolvedP = this.resolveFilename(`open '${p}'`, p);
+
+    let bigint: boolean;
+    let persistent: boolean;
+    let interval: number;
+
+    let listener: WatchFileCallback;
+
+    switch (typeof a) {
+      case `function`: {
+        bigint = false;
+        persistent = true;
+        interval = 5007;
+
+        listener = a;
+      } break;
+
+      default: {
+        ({
+          bigint = false,
+          persistent = true,
+          interval = 5007,
+        } = a);
+
+        listener = b!;
+      } break;
+    }
+
+    let statWatcher = this.statWatchers.get(resolvedP);
+    if (typeof statWatcher === `undefined`) {
+      statWatcher = new CustomStatWatcher<PortablePath>(this, resolvedP, {bigint});
+
+      statWatcher.start();
+
+      this.statWatchers.set(resolvedP, statWatcher);
+    }
+
+    statWatcher.registerChangeListener(listener, {persistent, interval});
+
+    return statWatcher;
+  }
+
+  unwatchFile(p: PortablePath, cb?: WatchFileCallback) {
+    const resolvedP = this.resolveFilename(`open '${p}'`, p);
+
+    const statWatcher = this.statWatchers.get(resolvedP);
+    if (!statWatcher)
+      return;
+
+    if (typeof cb === `undefined`)
+      statWatcher.unregisterAllChangeListeners();
+    else
+      statWatcher.unregisterChangeListener(cb);
+
+    if (!statWatcher.hasChangeListeners()) {
+      statWatcher.stop();
+      this.statWatchers.delete(resolvedP);
+    }
+  }
```

Comment:
Same thing, we could abstract all this. With the example above, we'd just call:

```ts
unwatch(resolvedP, cb);
```

With `unwatch` being what `createStatWatcher` returned.